clang/lib/CodeGen/CGCUDANV.cpp

```diff
@@ -894,8 +896,7 @@ llvm::Function *CGNVCUDARuntime::makeModuleCtorFunction() {
   // thread safety of the loaded program. Therefore we can assume sequential
   // execution of constructor functions here.
   if (IsHIP) {
-    auto Linkage = CudaGpuBinary ? llvm::GlobalValue::InternalLinkage
-                                 : llvm::GlobalValue::ExternalLinkage;
+    auto Linkage = llvm::GlobalValue::InternalLinkage;
```

Comment:
CudaGpuBinary is not always true.

For normal `-fgpu-rdc` usage, CudaGpuBinary is false here.

I think the proper fix is 

```
auto Linkage = RelocatableDeviceCode? llvm::GlobalValue::ExternalLinkage
                                 : llvm::GlobalValue::InternalLinkage;

```